azurerm/internal/services/devspace/parse/devspace_controller_test.go

```diff
@@ -0,0 +1,73 @@
+package parse
+
+import (
+	"testing"
+)
+
+func TestDevSpaceControllerId(t *testing.T) {
+	testData := []struct {
+		Name     string
+		Input    string
+		Expected *DevSpaceControllerId
+	}{
+		{
+			Name:     "Empty",
+			Input:    "",
+			Expected: nil,
+		},
+		{
+			Name:     "No Resource Groups Segment",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000",
+			Expected: nil,
+		},
+		{
+			Name:     "No Resource Groups Value",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/",
+			Expected: nil,
+		},
+		{
+			Name:     "Resource Group ID",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/foo/",
+			Expected: nil,
+		},
+		{
+			Name:     "Missing Controllers Value",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.DevSpaces/controllers/",
+			Expected: nil,
+		},
+		{
+			Name:  "DevSpace Controller ID",
+			Input: "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.DevSpaces/controllers/Controller1",
+			Expected: &DevSpaceControllerId{
+				Name:          "Controller1",
+				ResourceGroup: "resGroup1",
+			},
+		},
+		{
+			Name:     "Wrong Casing",
+			Input:    "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups/resGroup1/providers/Microsoft.DevSpaces/Controllers/",
```

Comment:
Should have a value on there also or will fail on missing value also, possibly hiding a condition?

Reply:
Oooh..good catch. TY - I also think you've answered something I was wondering about!